batch_test.go

```diff
@@ -98,18 +90,51 @@ func TestError(t *testing.T) {
 		})
 	}
 
-	close(w)
+	w <- struct{}{}
 
-	for i, errCh := range errChs {
+	for _, errCh := range errChs {
 		err := <-errCh
-		// operation 2 runs in the second batch, along with operations 1 and 3, which
-		// all fail since the whole batch fails because of operation 2.
-		if i == 0 && err != nil {
-			t.Error(err)
-		} else if i > 0 {
-			if got, want := err.Error(), "test error 2"; got != want {
-				t.Errorf("expecting %q got %q", want, err)
+		if got, want := err.Error(), "test error 2"; got != want {
+			t.Errorf("expecting %q got %q", want, err)
+		}
+	}
+}
+
+// TestFirstBatch ensures the initial batch accepts multiple items
+func TestFirstBatch(t *testing.T) {
+	n := 0
+
+	w := make(chan struct{})
+	defer close(w)
+
+	newJob := func() batchelor.Batch {
+		n++
+		return &heavyLifting{id: n, waiter: w}
+	}
+	q := batchelor.NewQueue(newJob)
+
+	for j := 1; j < 4; j++ {
+		opsPerBatch := j * 2
+		var processed int
+		errChs := make([]<-chan error, opsPerBatch)
+		for i := 0; i < len(errChs); i++ {
```

Comment:
you probably want `cap` instead of `len`: `len` of a channel returns the number of items it contains, not its size

(see https://play.golang.org/p/MlIwECZZqHp)